### DISCUSSION TITLE
❓ EfficientAD does not support input sizes smaller than 256.

### DISCUSSION THREAD

**Discussion by yjszyd0813** [open-edge-platform/anomalib#1723] [Q&A] [answered]
### What is the motivation for this task?

I have checked the structure of EfficientAD and it does not support input sizes smaller than 256. Would you like to consider improving this issue? This is caused by the Encoder in AutoEncoder. In practical applications, there may be images with dimensions smaller than 256.

### Describe the solution you'd like

![image](https://github.com/openvinotoolkit/anomalib/assets/80048995/ebe68ce0-6e4e-4f40-af64-b069fa774205)

**Comment by blaz-r** [comment, marked as answer]
Hello. I believe that this is sadly not viable. The structure is defined by the authors of EfficientAD, so changing it would probably affect the performance.

**Comment by yjszyd0813** [comment]
Thank you for your reply. I understand what you mean. I think if we maintain the same approach as the author and redesign a network structure, we should also achieve good performance. Do you think this can be successful?

**Comment by blaz-r** [comment]
I think it could work, probably maintaining some of the good performance, but I can't say for sure.

**Reply by watertianyi** [reply]
Is there any solution to this now?

**Reply by Likhith-Sugganahalli** [reply]
@watertianyi @blaz-r 

In cases like this, would centre padding the image with (0,0,0) values not work?

**Reply by alexriedel1** [reply]
> @watertianyi @blaz-r
> 
> In cases like this, would centre padding the image with (0,0,0) values not work?

sure, but the easiest way would be to resize your images to 256,256
there is also a good chance that this will improve your prediction accuracy
